samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -415,20 +407,34 @@ public void stop() {
 
 
   /**
-   * Checks if a request has expired.
+   * Checks if a request has expired. If this request is due to a control action, then expiry timeout can be overridden
+   * otherwise defaults to
+   *
    * @param request the request to check
    * @return true if request has expired
    */
   private boolean isRequestExpired(SamzaResourceRequest request) {
     long currTime = Instant.now().toEpochMilli();
-    boolean requestExpired =  currTime - request.getRequestTimestamp().toEpochMilli() > requestExpiryTimeout;
+    boolean requestExpired =  currTime - request.getRequestTimestamp().toEpochMilli() > getRequestTimeout(request);
     if (requestExpired) {
       LOG.info("Request for Processor ID: {} on host: {} with creation time: {} has expired at current time: {} after timeout: {} ms.",
           request.getProcessorId(), request.getPreferredHost(), request.getRequestTimestamp(), currTime, requestExpiryTimeout);
     }
     return requestExpired;
   }
 
+  /**
+   * Control actions like move, restarts can optionally override request expiry timeout. Otherwise it defaults to
+   * { @code requestExpiryTimeout }
+   */
+  private Long getRequestTimeout(SamzaResourceRequest request) {
+    Optional<Long> controlActionRequestExpirytimeout = containerManager.getActionExpiryTimeout(request.getProcessorId());
```

Comment:
@rmatharu this is the way I offer flexibility to do per-request expiry timeout, let me know your thoughts. If the timeout is present that takes higher precedence than one defined by `cluster-manager.container.
request.timeout.ms`

This timeout is not only used as a wait time when active container stop is issued

Reply:
Do we really need this override-able timeout? Or is it a nice to have that we've thought of?
I'd suggest keeping this out for now and just using the configured value.